modules/understand-etcd-performance.adoc

```diff
@@ -0,0 +1,119 @@
+// Module included in the following assemblies:
+// etcd/etcd-overview.adoc
+
+:_mod-docs-content-type: CONCEPT
+[id="understand-etcd-performance_{context}"]
+= Understanding etcd performance
+
+As a consistent distributed key-value store operating as a cluster of replicated nodes, etcd follows the Raft algorithm by electing one node as the leader and the others as followers. The leader maintains the system's current state and ensures that the followers are up-to-date.
+
+The leader node is responsible for log replication. It handles incoming write transactions from the client and writes a Raft log entry that it then broadcasts to the followers. 
+
+When an etcd client such as `kube-apiserver` connects to an etcd member that is requesting an action that requires a quorum, such as writing a value, if the etcd member is a follower, it returns a message indicating that the transaction should go to the leader. 
+
+When the etcd client requests an action that requires a quorum, such as writing a value, from the leader, the leader maintains the client connection open while it writes the local Raft log, broadcasts the log to the followers, and waits for the majority of the followers to acknowledge to have committed the log without failures. The leader sends the acknowledgment to the etcd client and closes the session. If failure notifications are received from the clients and fail to reach a consensus, the leader returns the error message to the client and closes the session.
+
+.{product-title} timer conditions for etcd
+
+{product-title} maintains etcd timers that are optimized for each platform. {product-title} has prescribed validated values that are optimized for each platform provider. The default `etcd timers` with `platform=none` or `platform=metal` are as follows:
+
+[source,yaml]
+----
+- name: ETCD_ELECTION_TIMEOUT <1>
+  value: "1000"
+  ...
+- name: ETCD_HEARTBEAT_INTERVAL <2>
+  value: "100" 
+----
+
+<1> This timeout is how long a follower node will go without hearing a heartbeat before it attempts to become the leader.
+<2> The frequency that the leader notifies followers that it is still the leader.
+
+These parameters do not provide all of the information for the control plane or for etcd. An etcd cluster is sensitive to disk latencies. Because etcd must persist proposals to its log, disk activity from other processes might cause long `fsync` latencies. The consequence is that etcd might miss heartbeats, causing request timeouts and temporary leader loss. During a leader loss and reelection, the Kubernetes API cannot process any request that causes a service-affecting event and instability of the cluster.
+
+.Effects of disk latency on etcd
+
+An etcd cluster is sensitive to disk latencies. To understand the disk latency that is experienced by etcd in your control plane environment, run the Flexible I/O Tester (fio) tests or suite, to check etcd disk performance in {product-title}.
```

Comment:
Suggestion: Maybe a quick note on the fact that `fio` is a point in time test that only tells whether the disk latency was fast enough at the time `fio` is executed and does not account for long term disk behaviour and other disk workloads that will be present alongside etcd in a real scenario.

It's mentioned in the KCS but maybe worth surfacing here as a quick mention.
https://access.redhat.com/solutions/4885641

But if that's too much info for an overview then we can skip it.

Reply:
@hasbro17  I added an important note here.